src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/SourceInfo.cs

```diff
@@ -58,12 +61,35 @@ internal class SourceInfo : ITaintedDataInfo, IEquatable<SourceInfo>
         /// <summary>
         /// Methods that generate tainted data and whose arguments don't need extra value content analysis.
         /// </summary>
-        public ImmutableDictionary<string, IsInvocationTaintedWithPointsToAnalysis> TaintedMethodsNeedPointsToAnalysis { get; }
+        /// <remarks>
+        /// MethodMatcher determines if the outermost tuple applies, based on the method names and arguments.
+        /// PointsToCheck determines if the inner tuple applies, based on the method invocation's arguments PointsToAbstractvalues.
+        /// TaintedTarget is the tainted target (arguments / return value).
+        ///
+        /// Example:
+        /// (
+        ///   (methodName, argumentOperations) => methodName == "Bar",  // MethodMatcher
+        ///   {
+        ///      (
+        ///         (pointsTos) => true,  // PointsToCheck
+        ///         TaintedTargetValue.Return  // TaintedTarget
+        ///      )
+        ///   }
+        /// )
+        ///
+        /// will treat any invocation of the "Bar" method's return value as tainted.
+        /// </remarks>
+        public ImmutableHashSet<(MethodMatcher, ImmutableHashSet<(PointsToCheck, string)>)> TaintedMethodsNeedsPointsToAnalysis { get; }
```

Comment:
>string [](start = 81, length = 6)

If you name the tuple's items, it'll be be more meaningful than just the type (especially if the type is string).

```csharp
ImmutableHashSet<(MethodMapper MethodMapper, ImmutableHeshSet<(PointsToCheck PointsToCheck, string TaintedTarget)>>
```